src/charmed_kubeflow_chisme/bundle/_bundle.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Helper to work with Juju bundles, preserving comments in the YAML."""
+
+from __future__ import annotations
+
+import copy
+from pathlib import Path
+from typing import Optional
+
+from deepdiff import DeepDiff
+from ruamel.yaml import YAML
+
+from ..juju import Juju
+
+
+class Bundle:
+    """Juju bundle loader/dumper.
+
+    This class uses ruamel.yaml instead of the typical pyyaml in order to preserve all comments,
+    the order of the yaml, etc.  This way we can load/dump a yaml with comments without losing them
+    or reordering the items.
+    """
+
+    def __init__(self, filename: Optional[str] = None):
+        self._filename = filename
+        self._data = None
+
+        # Auto-load the bundle if a filename is provided
+        if self._filename:
+            self._load_bundle()
```

Comment:
You changed the call here but not the method name (its still called load_yaml). I wanted to change it but not sure if sth else from outside is not calling load_bundle

Reply:
good catch.  Renamed the method to _load_bundle